src/storage/v2/vertex_accessor.hpp

```diff
@@ -68,6 +68,11 @@ class VertexAccessor final {
   /// @throw std::bad_alloc
   Result<PropertyValue> SetProperty(PropertyId property, const PropertyValue &value);
 
+  /// Set property values only if property store is empty. Returns `true` if successully set all values,
+  /// `false` otherwise.
+  /// @throw std::bad_alloc
+  Result<bool> SetProperties(std::map<storage::PropertyId, storage::PropertyValue> &properties);
```

Comment:
What do you mean why not just set the properties? 

I know you probably know how `PropertyStore` works, but I just want to write it here in any case. 

I have implemented `SetProperties` method in such way because when we are setting all new properties (on vertex creation and edge creation) there doesn't have to be check in property store if a property exists. This check proved to be expensive when we are setting a lot of properties on a lot of nodes. What happens when we want to set for example 50 properties with the old way? First, you set 1 property. Then when you want to set the next property you need to check if the such property exists. Although you as a user of property store methods know that such property is not there yet, the property store will make such a check. And it will do for every new property you want to set. 

Constraint here is that the property store needs to be empty so we can skip checks if property exists.

Reply:
And furthermore, if some properties are already stored in property store, due to the way the property store works now by ordering properties by their property key in ascending order, we should also make sure that is aligned, which would mean either first reading all the currently stored properties to the `std::map` and add to such map also new properties, which would guarantee ordering when reading from it in ascending order and then store in property store.
Second option in such case if some properties already exist is same as what is happening now. Find place where to store new property, copy memory around that property, possibly prepare new memory block and delete old one if necessary.

Reply:
Okay, so the reason is it would be to complex and we doesn't need that to achieve what we want, then it is perfect as is!